app/org/sagebionetworks/bridge/upload/UploadValidationTask.java

```diff
@@ -111,6 +124,30 @@ public void run() {
         }
 
         // TODO: if validation fails, wipe the files from S3
+
+        // dedupe logic over here:
+        dedupeHelper(context.getRecordId());
+    }
+
+    /**
+     * helper method: query healthdatarecord by healthcode, schema and createdOn from the upload just finished above,
+     * if there are more than 1 such records in ddb, there are duplicates. log that information but not stop or delete anything right now
+     * @param healthRecordId
+     */
+    private void dedupeHelper(String healthRecordId) {
+        // get necessary information for query first
+        HealthDataRecord record = healthDataService.getRecordById(healthRecordId);
```

Comment:
There might not be a record for this upload. (For example, if the upload failed validation.) What does your code do in this case?